File: src/_027/kotlin/Solution.kt
package _027.kotlin

/**
 * @author relish
 * @since 2018/04/13
 */
class Solution {
    fun removeElement(nums: IntArray, `val`: Int): Int {
        var count = 0
        for (index in nums.indices) {
            val t = nums[index]
            if (t != `val`) {
                nums[count++] = t
            }
        }
        return count
    }
}

fun main(args: Array<String>) {
    val arr = intArrayOf(3, 2, 3, 1, 2, 3, 2)
    println(Solution().removeElement(arr, 3))
}

File: src/_027/Solution.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    var len = nums.length
    for(var i = 0; i < len; i++) {
        if (nums[i] === val) {
            nums.splice(i, 1)
            i = i - 1
        }
    }
    return nums.length
};
